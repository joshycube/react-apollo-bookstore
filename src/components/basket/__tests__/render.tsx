import React from "react";
import BasketRender, { getTotal } from "../render";

describe("Basket render", () => {
  let props: any;

  beforeEach(() => {
    props = {
      basket: {
        items: [
          {
            bookId: 1,
            title: "foo",
            author: "me",
            price: 12.5
          },
          {
            bookId: 2,
            title: "bar",
            author: "me",
            price: 1.99
          }
        ]
      }
    };
  });

  it("should render the sum and total right", () => {
    const wrapper = shallow(<BasketRender {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("getTotal", () => {
  const items = [
    {
      bookId: 1,
      title: "foo",
      author: "me",
      price: 12.5
    },
    {
      bookId: 2,
      title: "bar",
      author: "me",
      price: 1.99
    }
  ];
  it("should return the correct total", () => {
    const total = items[0].price + items[1].price;
    expect(getTotal(items)).toEqual(total);
  });
});
