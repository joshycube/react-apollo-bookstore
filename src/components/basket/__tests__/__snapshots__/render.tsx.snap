// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket render should render the sum and total right 1`] = `"<div class=\\"sc-bdVaJa gdqIHp\\">You have selected: 2 books with the total price of: Â£14.49</div>"`;
